src/components/policy/src/policy/src/access_remote_impl.cc

```diff
@@ -405,12 +504,55 @@ bool AccessRemoteImpl::CheckPTUUpdatesChange(
     return false;
   }
 
-  if (new_consent.is_initialized() && old_consent.is_initialized())
-    return *new_consent != *old_consent;
+  bool result = false;
+  if (new_consent.is_initialized() && old_consent.is_initialized()) {
+    result = (*new_consent != *old_consent);
+  } else {
+    bool not_changed_consent1 = !new_consent.is_initialized() && *old_consent;
+    bool not_changed_consent2 = !old_consent.is_initialized() && *new_consent;
+
+    result = !(not_changed_consent1 || not_changed_consent2);
+  }
 
-  bool not_changed_consent1 = !new_consent.is_initialized() && *old_consent;
-  bool not_changed_consent2 = !old_consent.is_initialized() && *new_consent;
-  return !(not_changed_consent1 || not_changed_consent2);
+  if (result) {
+    listener_->OnRemoteAllowedChanged(result);
```

Comment:
I don't understand. I think AccessRemote doesn't need to have listener. PM can invoke listener. AccessRemote as CachManager is only for calling and returning result for PM. You added the circular dependent: AccessRemote invokes PH, PH -> PM, PM -> AccessRemote. It is awful.